22_CppCompat/app_resources/common.hlsl

```diff
@@ -25,11 +28,352 @@
 
 #include <nbl/builtin/hlsl/limits.hlsl>
 
-#define DISABLE_TGMATH_TESTS
-#ifndef DISABLE_TGMATH_TESTS 
-#include <nbl/builtin/hlsl/cmath.hlsl>
-#endif
 
 #include <nbl/builtin/hlsl/barycentric/utils.hlsl>
 #include <nbl/builtin/hlsl/member_test_macros.hlsl>
 #include <nbl/builtin/hlsl/device_capabilities_traits.hlsl>
+
+#include <nbl/builtin/hlsl/tgmath.hlsl>
+#include <nbl/builtin/hlsl/cpp_compat/intrinsics.hlsl>
+
+// tgmath.hlsl and intrinsics.hlsl tests
+
+using namespace nbl::hlsl;
+struct TgmathIntputTestValues
+{
+	float floor;
+	float isnan;
+	float isinf;
+	float powX;
+	float powY;
+	float exp;
+	float exp2;
+	float log;
+	float log2;
+	float absF;
+	int absI;
+	float sqrt;
+	float sin;
+	float cos;
+	float acos;
+	float modf;
+	float round;
+	float roundEven;
+	float trunc;
+	float ceil;
+	float fmaX;
+	float fmaY;
+	float fmaZ;
+	float ldexpArg;
+	int ldexpExp;
+	float modfStruct;
+	float frexpStruct;
+
+	float32_t3 floorVec;
+	float32_t3 isnanVec;
+	float32_t3 isinfVec;
+	float32_t3 powXVec;
+	float32_t3 powYVec;
+	float32_t3 expVec;
+	float32_t3 exp2Vec;
+	float32_t3 logVec;
+	float32_t3 log2Vec;
+	float32_t3 absFVec;
+	int32_t3 absIVec;
+	float32_t3 sqrtVec;
+	float32_t3 sinVec;
+	float32_t3 cosVec;
+	float32_t3 acosVec;
+	float32_t3 modfVec;
+	float32_t3 roundVec;
+	float32_t3 roundEvenVec;
+	float32_t3 truncVec;
+	float32_t3 ceilVec;
+	float32_t3 fmaXVec;
+	float32_t3 fmaYVec;
+	float32_t3 fmaZVec;
+	float32_t3 ldexpArgVec;
+	int32_t3 ldexpExpVec;
+	float32_t3 modfStructVec;
+	float32_t3 frexpStructVec;
+};
+
+struct TgmathTestValues
+{
+	float floor;
+	int isnan;
+	int isinf;
+	float pow;
+	float exp;
+	float exp2;
+	float log;
+	float log2;
+	float absF;
+	int absI;
+	float sqrt;
+	float sin;
+	float cos;
+	float acos;
+	float modf;
+	float round;
+	float roundEven;
+	float trunc;
+	float ceil;
+	float fma;
+	float ldexp;
+
+	float32_t3 floorVec;
+#ifndef __HLSL_VERSION
+	nbl::hlsl::vector<int, 3> isnanVec;
+	nbl::hlsl::vector<int, 3> isinfVec;
+#else
+	vector<int, 3> isnanVec;
+	vector<int, 3> isinfVec;
+#endif
```

Comment:
leave a comment that we can't fix this because `using namespace nbl::hlsl` would cause ambiguous math functions below and we can't add a `nbl::hlsl` alias for the builtin hLSL vector type because of https://github.com/microsoft/DirectXShaderCompiler/issues/7035